### PR TITLE
refactor(core): introduce debugName optional arg to ReactiveNode

### PR DESCRIPTION
**Pull request by AleksanderBodurri**
This commit contains the changes to core/primitives that used to be in https://github.com/angular/angular/pull/57073 but were [moved for g3sync](https://github.com/angular/angular/pull/57073#pullrequestreview-2283691431)

### CODE REVIEW

**Review by dgp1130 — APPROVED** [review]

**Comment by AleksanderBodurri** [comment]
Awaiting g3sync run @alxhub

**Review by alxhub — APPROVED** [review]
LGTM - just needs a change to the commit message to use `refactor(core)` instead of `feat(core)`, because this is not a user visible feature.

File: packages/core/primitives/signals/src/graph.ts [hunk 1]
```diff
@@ -179,6 +179,11 @@ export interface ReactiveNode {
    * Called when a signal is read within this consumer.
    */
   consumerOnSignalRead(node: unknown): void;
+
+  /**
+   * A debug name for the reactive node. Used in Angular DevTools to identify the node.
+   */
+  debugName?: string;
```
Comment (alxhub) [inline packages/core/primitives/signals/src/graph.ts:186]:
```suggestion
  debugName: string|undefined;
```
and add a value of `undefined` to `REACTIVE_NODE`.

**Review by AndrewKushnir — APPROVED** [review]

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Comment by AndrewKushnir** [comment]
TESTED=No TGP required, see https://github.com/angular/angular/pull/57710#issuecomment-2372301933.

**Pull request closed without merging**